src/plugins/chart_expressions/expression_heatmap/public/components/heatmap_component.tsx

```diff
@@ -271,8 +271,13 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = memo(
 
     // adds a very small number to the max value to make sure the max value will be included
     const smattering = 0.00001;
-    const endValue =
-      (paletteParams?.range === 'number' ? paletteParams.rangeMax : max) + smattering;
+    let endValue = max + smattering;
+    if (paletteParams?.rangeMax) {
```

Comment:
```suggestion
    if (paletteParams?.rangeMax != null) {
```

I suspect in case values are negative and the endValue/`rangeMax` is `0` this check would fail. What do you think?